### ISSUE TITLE
Question: Why have distinct index buffers per attribute?

### ISSUE THREAD

**Issue by nicopap** [FractalFir/tmf#8]
The API exposes `set_normal_triangles`, `set_uv_triangles`, `set_vertex_triangles`. This allows user to use different index buffers per attribute.

In my experience, there is only a single set of indices per mesh. So all those `foo_triangles` arrays are duplicate of the same index buffer!

So why do they exist as separate entities? Shouldn't they be merged into a single one? It seems this could reduce memory usage as well.

**Comment by FractalFir** [comment]
Those **are not** copies of the same buffer. The each have different data and sizes. This allows the normal, uv, and vertex position buffers to be different sizes too. 

As an example, a mesh of voxels consists may consist of any number of points sharing UV coordinates. Without different buffers, the number of UV coordinates would have to match the number of voxel coordinates. In a case of a voxel mesh with 10 000 vertices, that would be at least 9 000 additional UV coordinates, which would ballon the file size. 

This also allows merging of some data(not all) of two unrelated vertices. Two vertices may share the same normal, but have different uvs and positions. If there was only one index buffer, there would be nothing that can be done. With 3 buffers, the normal may be saved only once, saving space.

**Comment by nicopap** [comment]
I see, I understand. This makes sense. I've not fiddled enough with rendering to see this use case, it's obvious in insight.

I think it's enough for this issue. I might have follow up questions later.
